src/dev/flang/be/c/Intrinsics.java

```diff
@@ -202,6 +202,67 @@ public class Intrinsics extends ANY
             );
         }
         );
+    put("fuzion.std.fileio.on_open"   , (c,cl,outer,in) ->
+        {
+          var filepointer = new CIdent("fp");
+          var openresults = new CIdent("open_results");
+          var errno = new CIdent("errno");
+          return CStmnt.seq(
+            CExpr.decl("long *", openresults),
+            openresults.assign(A1.castTo("long *")),
+            CExpr.decl("int", errno),
+            errno.assign(new CIdent("0")),
+            CStmnt.suitch(
+              A2.castTo("int"),
+              new List<>(
+                CStmnt.caze(
+                  new List<>(CExpr.int8const(0)),
+                  CStmnt.seq(
+                    CExpr.decl("FILE *", filepointer, CExpr.call("fopen", new List<>(A0.castTo("char *"), CExpr.string("r")))),
+                    CExpr.iff(CExpr.notEq(filepointer, new CIdent("NULL")),
+                      CStmnt.seq(openresults.index(CExpr.ident("0")).assign(filepointer.castTo("fzT_1i64")))),
+                    CStmnt.BREAK
+                    )
+                  ),
+                CStmnt.caze(
+                  new List<>(CExpr.int8const(1)),
+                  CStmnt.seq(
+                    CExpr.decl("FILE *", filepointer, CExpr.call("fopen", new List<>(A0.castTo("char *"), CExpr.string("w")))),
+                    CExpr.iff(CExpr.notEq(filepointer, new CIdent("NULL")),
+                      CStmnt.seq(openresults.index(CExpr.ident("0")).assign(filepointer.castTo("fzT_1i64")))),
+                    CStmnt.BREAK
+                    )
+                  ),
+                CStmnt.caze(
+                  new List<>(CExpr.int8const(2)),
+                  CStmnt.seq(
+                    CExpr.decl("FILE *", filepointer, CExpr.call("fopen", new List<>(A0.castTo("char *"), CExpr.string("a")))),
+                    CExpr.iff(CExpr.notEq(filepointer, new CIdent("NULL")),
+                      CStmnt.seq(openresults.index(CExpr.ident("0")).assign(filepointer.castTo("fzT_1i64")))),
+                    CStmnt.BREAK
+                    )
+                  )
+                ), 
+              CStmnt.seq(
+                CExpr.fprintfstderr("*** Unsupported open flag. Please use: 0 for READ, 1 for WRITE, 2 for APPEND. ***\n"),
+                CExpr.exit(1)
```

Comment:
`EDOM 33` `/* Math argument out of domain of func */`  is referred to as `Used by mathematical functions when an argument value does not fall into the domain over which the function is defined` in the GNU C Library so I am not sure whether `fopen` should be considered as a mathematical function.

Reply:
I think a hard exit is fine here since this is only in case an internal parameter is wrong. Should not happen for a parameter that user could provide, though.